network/src/peer.rs

```diff
@@ -582,35 +560,35 @@ impl<N: Network, E: Environment> Peer<N, E> {
                                     // Update the node type of the peer.
                                     peer.node_type = node_type;
                                     // Update the status of the peer.
-                                    peer.status.update(status);
```

Comment:
If we can directly assign `status`, do we even need the `update()` method? Can/should it be removed?

Reply:
with `Peer`, we have the option of mutating this field during synchronous access (so we can just assign to it directly), but for the immutable `Node` shared across multiple threads we still need to use `Status::update` that uses atomics